app/controllers/api/v1/base_controller.rb

```diff
@@ -8,33 +8,14 @@ class BaseController < ApplicationController
       private
 
       def require_authentication!
-        authenticator = Authenticator.new(
-          user: User.find_by(id: request.headers["X-User-Id"])
-        )
+        user = User.find_by(id: request.headers["X-User-Id"])
 
-        if authenticator.valid_token?(request.headers["Authorization"])
-          sign_in authenticator.user
+        if user&.valid_token?(request.headers["Authorization"])
```

Comment:
`&.` 😿

Reply:
gonna be tricky to get rid of this one 💃

Reply:
😿 And until Rails adds proper null object handling, I'll keep leaving crying cat faces in PR comments.

Reply:
Why not add a NullUser than :D

Reply:
It's really tricky to do it nicely.